pkg/envoy/eds/response.go

```diff
@@ -47,7 +47,15 @@ func getEndpointsForProxy(meshCatalog catalog.MeshCataloger, proxyIdentity ident
 			log.Error().Err(err).Msgf("Failed listing allowed endpoints for service %s for proxy identity %s", dstSvc, proxyIdentity)
 			continue
 		}
-		allowedServicesEndpoints[dstSvc] = endpoints
+		var availableEndpoints []endpoint.Endpoint
+		for _, endpoint := range endpoints {
+			if proxyRegistry.IsProxyConnected(certificate.CommonName(endpoint.ProxyUID + "." + dstSvc.Name + "." + dstSvc.Namespace)) {
+				availableEndpoints = append(availableEndpoints, endpoint)
+			}
+		}
+		if len(availableEndpoints) > 0 {
+			allowedServicesEndpoints[dstSvc] = availableEndpoints
+		}
```

Comment:
I wonder if we need to do this? I think it would be the same if we did `allowedServicesEndpoints[dstSvc] = availableEndpoints` without checking the size of the slice.